enterprise/dev/ci/internal/ci/cache_helpers.go

```diff
@@ -0,0 +1,14 @@
+package ci
+
+import "github.com/sourcegraph/sourcegraph/enterprise/dev/ci/internal/buildkite"
+
+func cacheYarn() buildkite.StepOpt {
```

Comment:
nit: verb "cache yarn" seems to imply leaving things behind for the next build, but this also loads from a matching cache right? Should it be called:

```suggestion
func withYarnCache() buildkite.StepOpt {
```

Reply:
That's much better semantics, thank you!